docker/install/ubuntu_install_qemu.sh

```diff
@@ -16,22 +16,46 @@
 # specific language governing permissions and limitations
 # under the License.
 
+#
+# Install QEMU on Ubuntu.
+#
+# Usage: ubuntu_install_qemu.sh [--target-list target0,target1,...]
+#   --target-list is list of target for QEMU comma seperated. e.g. aarch64-softmmu,arm-softmmu,...
+#
+
+set -e
+set -o pipefail
+
+QEMU_DIR=qemu-5.1.0
+
+# Get number of cores for build
 if [[ "${CI_NUM_CORE}" ]]; then
-  NUM_CORE=${CI_NUM_CORE}
+  num_cores=${CI_NUM_CORE}
 else
-  NUM_CORE=2
+  num_cores=2
 fi
 
-set -e
-set -u
-set -o pipefail
+# Set target list for QEMU
+if [ "$1" == "--target-list" ]; then
+    shift
+    target_list=$1
+else
+    target_list="aarch64-softmmu,arm-softmmu,i386-softmmu,riscv32-softmmu,riscv64-softmmu,x86_64-softmmu"
+fi
 
-sudo sed -i '/deb-src/s/^# //' /etc/apt/sources.list
-apt update
-apt-get -y build-dep qemu
+# Check if QEMU already built
+rebuild=0
```

Comment:
why do we need this logic in this file?

Reply:
The reason is that we could update the RVM on vagrant cloud with a version that has qemu already installed with limited targets. And then if someone is using this locally, it will only rebuild with new set of targets.

Reply:
ah i see. okay, do you intend to push that update soon? otherwise, may be better to keep this file simple. i agree it's a nice feature, but wondering how much extra it adds to the base image.